timber_test.go

```diff
@@ -61,6 +64,75 @@ func TestDoubleClose(t *testing.T) {
 		Level:     DEBUG,
 		Formatter: formatter})
 	log.Close()
-	log.Close() // call Close twice	
+	log.Close() // call Close twice
 	log.Warn("Don't panic")
 }
+
+// writer to capture logs for testing purposes
+type TestWriter struct {
+	logs []string
+}
+
+func (tw *TestWriter) LogWrite(msg string) {
+	tw.logs = append(tw.logs, msg)
+}
+
+func (tw *TestWriter) Close() {
+	// Nothing
+}
+
+func TestJSONFormatterLogger(t *testing.T) {
+	a := assert.New(t)
+
+	log := NewTimber()
+	testWriter := new(TestWriter)
+	formatter := NewJSONFormatter()
+	log.AddLogger(
+		ConfigLogger{
+			LogWriter: testWriter,
+			Level:     DEBUG,
+			Formatter: formatter,
+		},
+	)
+	log.Info("Some JSON logging")
+	log.InfoEx(
+		map[string]string{
+			"testExtra":        "hello",
+			"testAnotherExtra": "goodbye",
+		},
+		"Some JSON logging with some extra fields",
+	)
+	log.Close()
+
+	firstLog := testWriter.logs[0]
+	secondLog := testWriter.logs[1]
+
+	var firstLogMap map[string]interface{}
+	var secondLogMap map[string]interface{}
+
+	json.Unmarshal([]byte(firstLog), &firstLogMap)
+	json.Unmarshal([]byte(secondLog), &secondLogMap)
+
+	expectedSharedKeys := []string{"Level", "timestamp", "SourceFile", "SourceFile", "message", "FuncPath", "FuncPath", "PackagePath", "HostName"}
```

Comment:
Are the repeated keys (`SourceFile` and `FuncPath`)  intentional? The following code suggests we only test presence?

Reply:
:smile: 

Got a github notif and thought I'd check it out.

Hope you're all keeping well in Leeds.